cyclonedx/model/tool.py

```
@@ -0,0 +1,430 @@
+
+        Returns:
+            A sorted set of Tools
+        """
+        tools_to_render: 'SortedSet[Tool]' = SortedSet()
+
+        for c in components:
+            tools_to_render.add(Tool(
+                name=c.name,
+                vendor=c.group,
+                version=c.version,
+                hashes=c.hashes,
+                external_references=c.external_references,
+            ))
+
+        for s in services:
+            if s.provider:
+                vendor = s.provider.name
+            else:
+                vendor = None
+            tools_to_render.add(Tool(
+                name=s.name,
+                vendor=vendor,
+                version=s.version,
+                external_references=s.external_references,
+            ))
+
+        return tools_to_render
+
+    @classmethod
+    def json_normalize(cls, o: ToolsRepository, *,
+                       view: Optional[Type[ViewType]],
+                       **__: Any) -> Any:
+        if not any([o.tools, o.components, o.services]):
+            return None
+
+        result = {}
+
+        if (view().schema_version_enum >= SchemaVersion1Dot5().schema_version_enum  # type: ignore[union-attr, misc]
+                and not o.tools):
+            if o.components:
+                result['components'] = [json_loads(Component.as_json(c, view_=view))  # type: ignore[attr-defined]
+                                        for c in o.components]
+
+            if o.services:
+                result['services'] = [json_loads(Service.as_json(s, view_=view))  # type: ignore[attr-defined]
+                                      for s in o.services]
+
+            if result:
+                return result
+
+        tools_to_render: 'SortedSet[Tool]' = SortedSet(o.tools)
+        # We "down-convert" Components and Services to Tools so we can render to older schemas
+        # or when there are existing Tool objects
+        tools_to_render.update(cls.convert_new_to_old(o.components, o.services))
```

I'm still uncomfortable with this. If there are Tools and (Components or Services), I feel we should throw an exception since they shouldn't be combined in the first place.

---

will rework this